Packs/DSPM/Integrations/DSPM/DSPM.py

```diff
@@ -0,0 +1,903 @@
+        if not matches:
+            raise ValueError(f'This "{sort_order}" sorting order is not supported')
+
+    params = {
+        "region.in": args.get("regionIn"),
+        "region.equals": args.get("regionEqual"),
+        "cloudProvider.in": args.get("cloudProviderIn"),
+        "cloudProvider.equals": args.get("cloudProviderEqual"),
+        "serviceType.in": args.get("serviceTypeIn"),
+        "serviceType.equals": args.get("serviceTypeEqual"),
+        "digTagKey.contains": args.get("digTagKeyContains"),
+        "digTagValue.contains": args.get("digTagValueContains"),
+        "lifecycle.in": args.get("lifecycleIn"),
+        "lifecycle.equals": args.get("lifecycleEqual"),
+        "sort": args.get("sort"),
+        "page": page,
+        "size": MAX_PAGE_SIZE,
+    }
+    # Remove None values from params
+    params = {k: v for k, v in params.items() if v is not None}
+
+    # Fetch data from client
+    response = client.get_asset_lists(params)
+
+    assets = response.get("assets", []) if isinstance(response, dict) else response
+
+    if not assets:
+        return []  # No more assets to fetch
+
+    parsed_assets = [
+        {
+            "ID": asset.get("id", ""),
+            "Project ID": asset.get("projectId", ""),
+            "Project Name": asset.get("projectName", ""),
+            "Name": asset.get("name", ""),
+            "Cloud Provider": asset.get("cloudProvider", ""),
+            "Cloud Environment": asset.get("cloudEnvironment", ""),
+            "Service Type": asset.get("serviceType", ""),
+            "Lifecycle": asset.get("lifecycle", ""),
+            "Open Risks Count": asset.get("openRisksCount", 0),
+            "Open Alerts Count": asset.get("openAlertsCount", 0),
+            "Encrypted": asset.get("encrypted", False),
+            "Open To World": asset.get("openToWorld", False),
+            "Tags": asset.get("tags", {}),
+            "Asset Dig Tags": asset.get("assetDigTags", []),
+        }
+        for asset in assets
+    ]
+    return parsed_assets
+
+
+def get_asset_details(client: Client, args: dict[str, Any]) -> CommandResults:
+    asset_id = args.get("asset_id", None)
+    if not asset_id:
+        raise ValueError("asset_id not specified")
+
+    asset_details = client.get_asset_details(asset_id)
+    demisto.debug(f"Asset details of : {asset_id}")
+    demisto.debug(asset_details)
+    return CommandResults(
+        outputs_prefix="DSPM.AssetDetails",
+        outputs_key_field="id",
+        outputs=asset_details,
+    )
+
+
+def get_asset_files_by_id(client: Client, args: dict[str, Any]) -> CommandResults:
+    asset_id = args.get("asset_id", None)
+    if not asset_id:
+        raise ValueError("Asset ID not specified")
+
+    page_number = 1
+    page_size = MAX_PAGE_SIZE  # You can adjust the page size as needed
+    all_files = []
+
+    while True:
+        params = {
+            "id": asset_id,
+            "page": page_number,
```

Comment:
```suggestion
    all_files = []

    while True:
        params = {
            "id": asset_id,
            "page": MAX_PAGE_SIZE,
```

Reply:
I believe your suggesting to assign the MAX_PAGE_SIZE constant to the size parameter.

Reply:
my bad, please revert this code